tests/tck-build-logic/src/main/java/org/graalvm/internal/tck/TestedVersionUpdaterTask.java

```diff
@@ -0,0 +1,73 @@
+package org.graalvm.internal.tck;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.core.util.DefaultIndenter;
+import com.fasterxml.jackson.core.util.DefaultPrettyPrinter;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.SerializationFeature;
+import org.graalvm.internal.tck.model.MetadataVersionsIndexEntry;
+import org.gradle.api.DefaultTask;
+import org.gradle.api.tasks.TaskAction;
+import org.gradle.api.tasks.options.Option;
+import org.gradle.util.internal.VersionNumber;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.List;
+
+public abstract class TestedVersionUpdaterTask extends DefaultTask {
+
+    private final ObjectMapper objectMapper = new ObjectMapper()
+            .enable(SerializationFeature.INDENT_OUTPUT)
+            .setSerializationInclusion(JsonInclude.Include.NON_NULL);
+
+    @Option(option = "coordinates", description = "GAV coordinates of the library")
+    void setCoordinates(String coordinates) {
+        this.coordinates = coordinates;
+    }
+
+    @Option(option = "lastSupportedVersion", description = "Last version of the library that passed tests")
+    void setLastSupportedVersion(String version) {
+        this.lastSupportedVersion = version;
+    }
+
+    private String coordinates;
+    private String lastSupportedVersion;
+
+
+    @SuppressWarnings("unchecked")
+    @TaskAction
+    void run() throws IllegalStateException, IOException {
+        List<String> GAVCoordinates = Arrays.stream(coordinates.split(":")).toList();
+        if (GAVCoordinates.size() != 3) {
+            throw new IllegalArgumentException("Maven coordinates should have 3 parts");
+        }
+
+        String group = GAVCoordinates.get(0);
+        String artifact = GAVCoordinates.get(1);
+        String version = GAVCoordinates.get(2);
```

Comment:
You found another way to do the same thing ;) 
```suggestion
        String[] coordinatesParts = coordinates.split(":");
        if (coordinatesParts.length != 3) {
            throw new IllegalArgumentException("Maven coordinates should have 3 parts");
        }

        String group = coordinatesParts[0];
        String artifact = coordinatesParts[1];
        String version = coordinatesParts[2];
```